src/shared/reportIssue.ts

```diff
@@ -0,0 +1,142 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Extension, vscode } from "../vscodeAdapter";
+import { Options } from "../shared/options";
+import { IHostExecutableResolver } from "../shared/constants/IHostExecutableResolver";
+import { basename, dirname } from "path";
+import { DotnetInfo } from "./utils/getDotnetInfo";
+
+const issuesUrl = "https://github.com/dotnet/vscode-csharp/issues/new";
+
+export default async function reportIssue(
+    vscode: vscode,
+    csharpExtVersion: string,
+    getDotnetInfo: (dotNetCliPaths: string[]) => Promise<DotnetInfo>,
+    shouldIncludeMonoInfo: boolean,
+    options: Options,
+    dotnetResolver: IHostExecutableResolver,
+    monoResolver?: IHostExecutableResolver) {
+    // Get info for the dotnet that the language server executable is run on, not the dotnet the language server will execute user code on.
+    let fullDotnetInfo: string | undefined;
+    try {
+        const info = await dotnetResolver.getHostExecutableInfo(options);
+        const dotnetInfo = await getDotnetInfo([ dirname(info.path) ]);
+        fullDotnetInfo = dotnetInfo.FullInfo;
+    } catch (error) {
+        const message = error instanceof Error ? error.message : `${error}`;
+        fullDotnetInfo = message;
+    }
+
+    let monoInfo = "";
+    if (shouldIncludeMonoInfo && monoResolver) {
+        monoInfo = await getMonoIfPlatformValid(options, monoResolver);
+    }
+    
+    let extensions = getInstalledExtensions(vscode);
+
+    const useOmnisharp = options.commonOptions.useOmnisharpServer;
+    const logInfo = getLogInfo(useOmnisharp);
+
+    const body = `## Issue Description ##
+## Steps to Reproduce ##
+
+## Expected Behavior ##
+
+## Actual Behavior ##
+
+## Logs ##
+
+${logInfo}
+
+## Environment information ##
+
+**VSCode version**: ${vscode.version}
+**C# Extension**: ${csharpExtVersion}
+**Using OmniSharp**: ${useOmnisharp}
+
+${monoInfo}
+<details><summary>Dotnet Information</summary>
+${fullDotnetInfo}</details>
+<details><summary>Visual Studio Code Extensions</summary>
+${generateExtensionTable(extensions)}
+</details>
+`;
+
+    const queryStringPrefix: string = "?";
```

Comment:
Why not just inline this?

Reply:
I didn't change this part (despite what git says).  I can make this better though.